File: FitsMeBack/src/main/java/FitsMe/FitsMe/Services/OutfitService.java
package FitsMe.FitsMe.Services;


import FitsMe.FitsMe.Dtos.OutfitsDto.OutfitRequest;
import FitsMe.FitsMe.Dtos.OutfitsDto.OutfitResponse;
import FitsMe.FitsMe.Dtos.UsuariosDtos.LogInRequest;
import FitsMe.FitsMe.Dtos.UsuariosDtos.UsuarioRequest;
import FitsMe.FitsMe.Dtos.UsuariosDtos.UsuarioResponse;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface OutfitService {


    OutfitResponse getOutfitbyId(Long outfitId);

    List<OutfitResponse> getOutfits();
    boolean DeleteOutfit(Long outfitId);
    OutfitResponse saveOutfit(OutfitRequest outfitRequest);
    List<OutfitResponse> getOutfitsByUserId(Long userId);

    List<OutfitResponse> getFavoriteOutfitsByUserId(Long userId);

    boolean setOutfitFavorito(Long outfitId , Long userId);

}

File: FitsMe/src/app/Services/OutfitService/outfit.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Outfit, OutfitRequest } from '../../Models/Models';

@Injectable({
  providedIn: 'root'
})
export class OutfitService {

  constructor(private http : HttpClient) { }
  selectedOutfit : Outfit = new Outfit();
  getOutfits()
  {
    return this.http.get('http://localhost:8080/Outfit/GetAllOutfits');
  }
  saveOutfit(outfit:OutfitRequest)
  {
    return this.http.post('http://localhost:8080/Outfit/SaveOutfit',outfit);
  }
  getOutfitsByUserId(userId:number){
    return this.http.get('http://localhost:8080/Outfit/GetOutfitsByUserId?userId='+ userId);
  }
  setOutfitFavorite(outfitId:number , userId:number)
  {
    return this.http.get("http://localhost:8080/Outfit/SetOutfitFavorito?outfitId=" + outfitId + "&userId=" + userId)
  }
  deleteOutfit(id:number)
  {
    return this.http.delete("http://localhost:8080/Outfit/DeleteOutfit?outfitId=" + id)
  }
  setOutfit(outfit:Outfit)
  {
    this.selectedOutfit = outfit;
  }
  getFavoriteOutfitsByUserId(userId:number){
    return this.http.get('http://localhost:8080/Outfit/GetFavoriteOutfitsByUserId?userId='+ userId);
  }
} 
